subprojects/docs/src/docs/userguide/authoring-builds/build_lifecycle.adoc

```diff
@@ -57,76 +73,81 @@ include::{snippetsPath}/buildlifecycle/basic/tests/buildlifecycle.out[]
 
-[[sec:configuration_and_execution_of_a_single_project_build]]
-== Configuration and execution of a single project build
+During settings file evaluation, Gradle:
 
-For a single project build, the workflow of the _after initialization_ phases are pretty simple. The build script is executed against the project object that was created during the initialization phase. Then Gradle looks for tasks with names equal to those passed as command line arguments. If these task names exist, they are executed as a separate build in the order you have passed them. The configuration and execution for multi-project builds is discussed in <<multi_project_configuration_and_execution.adoc#configuration_and_execution,Configuration and Execution>>.
+- Adds libraries to your build script classpath.
+- Defines which projects participate in a multi-project build.
 
-[[build_lifecycle_events]]
-== Responding to the lifecycle in the build script
+Gradle creates a `Project` for every project in the build.
+By default, each `Project` has a name equal to the name of its top level directory.
+Every project except the root project has a parent project.
+Any project may have child projects.
 
-Your build script can receive notifications as the build progresses through its lifecycle. These notifications generally take two forms: You can either implement a particular listener interface, or you can provide a closure to execute when the notification is fired. The examples below use closures. For details on how to use the listener interfaces, refer to the API documentation.
+== Configuration
 
+In the configuration phase, Gradle adds tasks and other properties to the projects generated by the initialization phase.
+By the end of the configuration phase, Gradle has a complete task execution graph for each project.
 
 [[sec:project_evaluation]]
-=== Project evaluation
+=== Project Evaluation
+
+During project evaluation, Gradle evaluates build scripts to build a task hierarchy within a `Project`.
+This hierarchy includes inputs, actions, and outputs for all tasks.
 
-You can receive a notification immediately before and after a project is evaluated. This can be used to do things like performing additional configuration once all the definitions in a build script have been applied, or for some custom logging or profiling.
+==== React to Project Evaluation
 
-Below is an example which adds a `test` task to each project which has a `hasTests` property value of true.
+You can receive a notification immediately before and immediately after a project evaluates.
+These notifications work even when project evaluation fails.
+You can configure project evaluation notifications for all projects or a specific project.
+For example, you could use these notifications for:
+
+- additional configuration after applying all definitions in a build script
+- custom logging
+- custom profiling
+
+The following example adds a `test` task to each project where the `hasTests` property value is `true`:
```

Comment:
This example uses `allprojects {}` which should be discouraged in favor of convention plugins.
This example also uses extra properties which should be discouraged in favor of project extensions registered by convention plugins.
This might be out of scope of this PR though.

Reply:
Interesting! This PR: https://github.com/gradle/gradle/issues/21216 is also on my list, which touches on this same issue... I'll DM you about how we can set a better example.